code/headers/display_adapter.hpp

```diff
@@ -61,13 +89,282 @@ namespace r2d2::display {
+         * @param filled a boolean which if true will create a filled circle and
+         * if false it will create a hollow circle
+         */
+        virtual void set_pixels_circle(uint8_t cursor_target, uint16_t radius,
+                                       bool filled) {
+            // Checks if the given cursor is not out of bounds
+            if (cursor_target >= CursorCount) {
+                return;
+            }
+            display_cursor_s cursor = cursors[cursor_target];
+
+            int x = cursor.cursor_x;
+            int y = cursor.cursor_y;
+            int color = color_to_pixel(cursor.cursor_color);
+
+            int t_x = radius;
+            int t_y = 0;
+            int err = 0;
+
+            if (filled == true) {
+                int xChange = 1 - (radius << 1);
+                int yChange = 0;
+
+                while (t_x >= t_y) {
+                    for (int i = x - t_x; i <= x + t_x; i++) {
+                        set_pixel(i, y + t_y, color);
+                        set_pixel(i, y - t_y, color);
+                    }
+                    for (int i = x - t_y; i <= x + t_y; i++) {
+                        set_pixel(i, y + t_x, color);
+                        set_pixel(i, y - t_x, color);
+                    }
+
+                    t_y++;
+                    err += yChange;
+                    yChange += 2;
+                    if (((err << 1) + xChange) > 0) {
+                        t_x--;
+                        err += xChange;
+                        xChange += 2;
+                    }
+                }
+            } else {
+
+                while (t_x >= t_y) {
+                    set_pixel(x + t_x, y + t_y, color);
+                    set_pixel(x + t_y, y + t_x, color);
+                    set_pixel(x - t_y, y + t_x, color);
+                    set_pixel(x - t_x, y + t_y, color);
+                    set_pixel(x - t_x, y - t_y, color);
+                    set_pixel(x - t_y, y - t_x, color);
+                    set_pixel(x + t_y, y - t_x, color);
+                    set_pixel(x + t_x, y - t_y, color);
+
+                    if (err <= 0) {
+                        t_y += 1;
+                        err += 2 * t_y + 1;
+                    }
+
+                    if (err > 0) {
+                        t_x -= 1;
+                        err -= 2 * t_x + 1;
+                    }
+                }
+            }
+        }
+
+        /**
+         * @brief Sets the targeted cursor to the given position. Can't go
+         * out of the size of the window. It will stop before that. Any
+         * characters written to be out of bounds will be drawn over each
+         * other.
+         *
+         * @param cursor_target This targets which cursor to move
+         * @param x new X position of the cursor
+         * @param y new Y position of the cursor
+         */
+        virtual void set_cursor_positon(uint8_t cursor_target, uint8_t x,
```

Comment:
Misspelled, should be  `set_cursor_position`

Reply:
should allready be fixed in the latest version